src/utils/validations.js

```diff
@@ -194,3 +194,38 @@ export const validateBmcURL = value => {
 
   return validateURL(value);
 };
+
+// Validates that the provided MAC address meets one of following formats supported by the golang ParseMAC function:
+// IEEE 802 MAC-48, EUI-48, EUI-64, or a 20-octet IP over InfiniBand link-layer address
+// https://golang.org/pkg/net/#ParseMAC
+export const isValidMAC = mac => {
+  if (mac.length < 14) {
+    return false;
+  }
+
+  let isValid = false;
+  if (mac[2] === ':') {
+    const regex = new RegExp(
+      '^(([0-9A-Fa-f]{2}[:]){19}[0-9A-Fa-f]{2})$|' + // 01:23:45:67:89:ab:cd:ef:00:00:01:23:45:67:89:ab:cd:ef:00:00
+      '^(([0-9A-Fa-f]{2}[:]){7}[0-9A-Fa-f]{2})$|' + // 01:23:45:67:89:ab:cd:ef
+        '^(([0-9A-Fa-f]{2}[:]){5}[0-9A-Fa-f]{2})$' // 01:23:45:67:89:ab
+    );
+    isValid = regex.test(mac);
+  } else if (mac[2] === '-') {
+    const regex = new RegExp(
+      '^(([0-9A-Fa-f]{2}[-]){19}[0-9A-Fa-f]{2})$|' + // 01-23-45-67-89-ab-cd-ef-00-00-01-23-45-67-89-ab-cd-ef-00-00
+      '^(([0-9A-Fa-f]{2}[-]){7}[0-9A-Fa-f]{2})$|' + // 01-23-45-67-89-ab-cd-ef
+        '^(([0-9A-Fa-f]{2}[-]){5}[0-9A-Fa-f]{2})$' // 01-23-45-67-89-ab
+    );
+    isValid = regex.test(mac);
+  } else if (mac[4] === '.') {
+    const regex = new RegExp(
+      '^(([0-9A-Fa-f]{4}.){9}[0-9A-Fa-f]{4})$|' + // 0123.4567.89ab.cdef.0000.0123.4567.89ab.cdef.0000
+      '^(([0-9A-Fa-f]{4}.){3}[0-9A-Fa-f]{4})$|' + // 0123.4567.89ab.cdef
+        '^(([0-9A-Fa-f]{4}.){2}[0-9A-Fa-f]{4})$' // 0123.4567.89ab
+    );
+    isValid = regex.test(mac);
```

Comment:
I would rather do either

```ts
return regex.test(mac)
```

or 
```ts
let regex;
...
regex = new RegExp(
```

and do test at the end

Reply:
Done! I went with the second option.